### ISSUE TITLE
Extend `@exposed` decorator to simplify and generalize recurring code checks

### ISSUE THREAD

**Issue by phorward** [viur-framework/viur-core#715]
This issue is a replacement for #618 to collect several scenarios and use-cases where an extended `@exposed`-decorator can be useful to configure a function's behavior, resulting in secure, less error prone and shorter code.

## `restrict`

To avoid function definitions like this:
```python
    @exposed
    def do_something(self, action: str):
        if not ((cuser := current.user.get()) and "root" in cuser["access"]):
            raise errors.Unauthorized()

        # ...
```
a `restrict`-attribute can be used like this:
```python
    @exposed(restrict="root")
    def do_something(self, action: str):
        # ...
```
or when thinking further, using a lambda:
```python
    @exposed(restrict=lambda user: "root" in user["access"] or "admin" in user["access"])
    def do_something(self, action: str):
        # ...
```

## `response`

To avoid function definitions like this:
```python
    @exposed
    def do_something(self, action: str):
        current.request.get().response.headers["Content-Type"] = "application/json"

        match action:
            # ...

            case _:
                raise errors.NotImplemented(f"Action {action!r} not implemented")

        return json.dumps("OKAY")
```
a `response`-attribute could be used like this.
```python
    @exposed(response="json")
    def do_something(self, action: str):
        match action:
            # ...

            case _:
                raise errors.NotImplemented(f"Action {action!r} not implemented")

        return "OKAY"
```

## `skey`

To avoid function definitions like this:
```python
    @exposed
    def do_something(self, action: str, skey: str):
        if not securitykey.validate(skey):
            raise errors.PreconditionFailed()
        # ...
```
an `skey`-attribute can be used,
```python
    @exposed(skey=True)
    def do_something(self, action: str):
        # ...
```
optionally with an lambda, evaluating the result of the `securitykey.validate()` function
```python
    @exposed(skey=lambda skey: skey["user"] == current.user.get()["key"])
    def do_something(self, action: str):
        # ...
```

**Comment by sveneberth** [comment]
In the ViUR meeting we decided to use multiple single decorators in `snake_case`.

**Comment by phorward** [comment]
Implemented by #800
